### PR TITLE
*: multi-query prefetch support pessimistic transaction

### PR DESCRIPTION
**Pull request by coocood**
### What problem does this PR solve?

Performance improvement.

### What is changed and how it works?

Call lock keys for pessimistic transactions for multi-statement query.
So we can avoid RPC call during each statement execution.

### Check List 

Tests 

- Unit test

### Release note 

- multi-query prefetch support pessimistic transaction

### CODE REVIEW

File: server/conn.go [hunk 1]
```diff
@@ -1330,10 +1330,14 @@ func (cc *clientConn) prefetchPointPlanKeys(stmts []ast.StmtNode) ([]plannercore
 	}
 	vars := cc.ctx.GetSessionVars()
 	if vars.TxnCtx.IsPessimistic {
-		// Pessimistic transaction do not benefit from the prefetch because the keys will be returned by Lock,
-		// we can get the keys from the lock cache.
-		// TODO: In the future we can support pre lock point keys for pessimistic transactions.
-		return nil, nil
+		if vars.IsReadConsistencyTxn() {
```
Comment (jackysp) [inline server/conn.go:1333]: I think if it is a multi-query, we could prefetch data for RC transaction.

Comment (coocood) [reply server/conn.go:1333]: We can do it in the future.

**Pull request merged**